src/hooks/api/useLunchMenu.ts

```diff
@@ -1,36 +1,62 @@
-import { useEffect, useState } from 'react';
-
-type LunchMenu = {
-  statusCode: number;
-  data: {
-    lunchMenu: { menu: string; allergy: number[] }[];
-    origins: { ingredient: string; origin: string }[];
-  };
-};
+/* eslint-disable react-hooks/exhaustive-deps */
+import { useCallback, useEffect, useState } from 'react';
+
+import { useUserInfo } from '@Hooks/useUserInfo';
+
+import { LunchMenu, LunchMenus } from '@Types/lunchMenu';
+
+const useLunchMenu = (date: Date, type: 'weekend' | 'day') => {
+  const { data } = useUserInfo();
+  const [isLoading, setIsLoading] = useState(true);
+  const [lunchMenu, setLunchMenu] = useState<LunchMenu[] | null>(null);
+  const [origins, setOrigins] = useState<[string, string][] | null>(null);
+
+  const fetchLunchMenu = useCallback(
+    async (areaCode: string, code: string) => {
+      setIsLoading(true);
 
-const useLunchMenu = () => {
-  const [lunchMenu, setLunchMenu] = useState<
-    undefined | { menu: string; allergy: number[] }[]
-  >();
-  const [isLoading, setIsLoading] = useState(false);
+      const token = localStorage.getItem('token');
 
-  const getLunchMenu = async () => {
-    setIsLoading(true);
+      const response = await fetch(
+        `/school/lunch-menu?areaCode=${areaCode}&schoolCode=${code}&date=${date}&type=${type}`,
+        {
+          headers: {
+            Authorization: `Bearer ${token}`,
+          },
+        },
+      );
 
-    const response = await fetch(
-      '/school/lunch-menu?areaCode=T10&schoolCode=9290083&date=230706',
-    );
-    const result = (await response.json()) as LunchMenu;
+      const { lunchMenu } = (await response.json()) as LunchMenus;
```

Comment:
서버에서 페칭한 데이터를 구조분해할당으로 받으면 만약 통신 오류가 나거나 데이터가 없어서 `undefined`일 때는 구조분해할당을 못 하니까 에러가 발생할 듯해요.
```suggestion
      const lunchMenu = (await response.json())?.lunchMenu;
```
이렇게 하면 `lunchMenu`가 `undefined`일 때 예외 처리를 할 수 있겠네요 :)

Reply:
맞아요! 현재는 무조건 성공한다는 보장하에 코드를 작성했어요. 즉, 에러핸들링이 없어요 😭 이것도 추후에 같이 이야기를 나누어봐요. 어떻게 통신 에러를 가져가야 할지요!

그리고 설탕쌤이 제안해주신 방법은 `lunchMenu`가 `any` 타입이여서 다음과 같이 리팩터링을 했습니다 :)

```typescript
const data = (await response.json()) as LunchMenus;

setLunchMenu(data.lunchMenu);
```